File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/mapping/RequeryMappingContextTest.java
package com.coupang.springframework.data.requery.mapping;

import com.coupang.springframework.data.requery.mapping.RequeryMappingContext;
import com.coupang.springframework.data.requery.mapping.RequeryPersistentEntity;
import io.requery.Key;
import io.requery.Version;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * com.coupang.springframework.data.requery.mapping.RequeryMappingContextTest
 *
 * @author debop
 * @since 18. 6. 8
 */
public class RequeryMappingContextTest {

    @Test
    public void requeryPersistentEntityRejectsSpringDataAtVersionAnnotation() {

        RequeryMappingContext context = new RequeryMappingContext();
        RequeryPersistentEntity entity = context.getRequiredPersistentEntity(Sample.class);

        assertThat(entity).isNotNull();

        assertThat(entity.getRequiredPersistentProperty("id").isIdProperty()).isTrue();
        assertThat(entity.getRequiredPersistentProperty("springId").isIdProperty()).isFalse();

        assertThat(entity.getRequiredPersistentProperty("version").isVersionProperty()).isTrue();
        assertThat(entity.getRequiredPersistentProperty("springVersion").isVersionProperty()).isFalse();
    }


    static class Sample {

        @Key
        long id;

        @org.springframework.data.annotation.Id
        long springId;

        @Version
        long version;

        @org.springframework.data.annotation.Version
        long springVersion;
    }
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/mapping/RequeryMappingContextTest.kt
package org.springframework.data.requery.mapping

import io.requery.Key
import io.requery.Version
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class RequeryMappingContextTest {

    @Test
    fun `persistent entity rejects Spring Data @Version annotation`() {

        val context = RequeryMappingContext()
        val entity = context.getRequiredPersistentEntity(Sample::class.java)

        assertThat(entity).isNotNull

        assertThat(entity.getRequiredPersistentProperty("id").isIdProperty).isTrue()
        assertThat(entity.getRequiredPersistentProperty("springId").isIdProperty).isFalse()

        assertThat(entity.getRequiredPersistentProperty("version").isVersionProperty).isTrue()
        assertThat(entity.getRequiredPersistentProperty("springVersion").isVersionProperty).isFalse()
    }

    private class Sample {
        
        @get:Key
        val id: Long? = 0

        @org.springframework.data.annotation.Id
        var springId: Long = 0

        @Version
        var version: Long = 0

        @org.springframework.data.annotation.Version
        var springVersion: Long = 0
    }
}